gui/layers/_markers_layer.py

```diff
@@ -260,16 +269,37 @@ def _slice_markers(self, indices):
         # Get a list of the coords for the markers in this slice
         coords = self.coords
         if len(coords) > 0:
-            matches = np.equal(
-                coords[:, 2:],
-                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
-
-            matches = np.all(matches, axis=1)
-
-            in_slice_markers = coords[matches, :2]
-            return in_slice_markers, matches
+            if self._n_dimensional is False:
+                matches = np.equal(
+                    coords[:, 2:],
+                    np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
+
+                matches = np.all(matches, axis=1)
+
+                in_slice_markers = coords[matches, :2]
+                return in_slice_markers, matches, 1
+            elif self.ndim > 2:
+                distances = abs(coords[:, 2:] -
+                                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
+                if isinstance(self.size, np.ndarray):
+                    if len(self.size) == len(indices):
+                        size_array = np.broadcast_to(self.size[2:]/2, (len(coords), len(indices) - 2))
+                    elif isinstance(self.size, np.ndarray) and len(self.size.shape)==2:
+                        size_array = self.size[:,2:]/2
+                    else:
+                        size_array = np.broadcast_to(self.size/2, (len(indices) - 2, len(coords))).T
+                else:
+                    size_array = np.broadcast_to(self.size/2, (len(indices) - 2, len(coords))).T
```

Comment:
There is a lot of repetition and branching logic that I don't understand here... Do you think it could be simplified? In situations like these I try to find a canonical expression that works (e.g. this last one, or maybe `size_array = self.size[:, 2:] / 2`) and then prepare the input so that it'll work with said expression (e.g. `self.size = np.broadcast_to(self.size / 2, (len(coords), len(indices) - 2))`. I'm sure I'm getting my shapes and sizes mixed up here but hopefully it is enough so that you get the idea...

Reply:
done